miden-tx/src/error.rs

```diff
@@ -166,151 +169,193 @@ impl fmt::Display for AuthenticationError {
-const ERR_NOTE_INVALID_SENDER: u32 = 131111;
-const ERR_NOTE_INVALID_ASSETS: u32 = 131112;
-const ERR_NOTE_INVALID_INPUTS: u32 = 131113;
-const ERR_NOTE_TOO_MANY_ASSETS: u32 = 131114;
-const ERR_VAULT_GET_BALANCE_WRONG_ASSET_TYPE: u32 = 131115;
-const ERR_VAULT_HAS_NON_FUNGIBLE_WRONG_ACCOUNT_TYPE: u32 = 131116;
-const ERR_VAULT_FUNGIBLE_MAX_AMOUNT_EXCEEDED: u32 = 131117;
-const ERR_VAULT_ADD_FUNGIBLE_ASSET_MISMATCH: u32 = 131118;
-const ERR_VAULT_NON_FUNGIBLE_ALREADY_EXISTS: u32 = 131119;
-const ERR_VAULT_FUNGIBLE_AMOUNT_UNDERFLOW: u32 = 131120;
-const ERR_VAULT_REMOVE_FUNGIBLE_ASSET_MISMATCH: u32 = 131121;
-const ERR_VAULT_NON_FUNGIBLE_MISSING_ASSET: u32 = 131122;
-const ERR_FUNGIBLE_ASSET_FORMAT_POSITION_ONE_MUST_BE_ZERO: u32 = 131123;
-const ERR_FUNGIBLE_ASSET_FORMAT_POSITION_TWO_MUST_BE_ZERO: u32 = 131124;
-const ERR_FUNGIBLE_ASSET_FORMAT_POSITION_THREE_MUST_BE_FUNGIBLE_FAUCET_ID: u32 = 131125;
-const ERR_FUNGIBLE_ASSET_FORMAT_POSITION_ZERO_MUST_BE_WITHIN_LIMITS: u32 = 131126;
-const ERR_NON_FUNGIBLE_ASSET_FORMAT_POSITION_ONE_MUST_BE_FUNGIBLE_FAUCET_ID: u32 = 131127;
-const ERR_NON_FUNGIBLE_ASSET_FORMAT_HIGH_BIT_MUST_BE_ZERO: u32 = 131128;
-const ERR_FUNGIBLE_ASSET_MISMATCH: u32 = 131129;
-const ERR_NON_FUNGIBLE_ASSET_MISMATCH: u32 = 131130;
-const ERR_ACCOUNT_NONCE_INCR_MUST_BE_U32: u32 = 131131;
-const ERR_ACCOUNT_INSUFFICIENT_ONES: u32 = 131132;
-const ERR_ACCOUNT_SET_CODE_ACCOUNT_MUST_BE_UPDATABLE: u32 = 131133;
-const ERR_ACCOUNT_SEED_DIGEST_MISMATCH: u32 = 131134;
-const ERR_ACCOUNT_INVALID_POW: u32 = 131135;
-const ERR_NOTE_DATA_MISMATCH: u32 = 131136;
-const ERR_ASSET_NOT_FUNGIBLE_ID: u32 = 131137;
-const ERR_ASSET_INVALID_AMOUNT: u32 = 131138;
-const ERR_ASSET_NOT_NON_FUNGIBLE_ID: u32 = 131139;
-const ERR_INVALID_NOTE_TYPE: u32 = 131140;
-const ERR_NOTE_INVALID_TAG_PREFIX_FOR_TYPE: u32 = 131141;
-const ERR_NOTE_TAG_MUST_BE_U32: u32 = 131142;
+// --- ACCOUNT ERRORS -----------------------------------------------------------------------------
+const ERR_ACCOUNT_CODE_MUST_BE_UPDATABLE: u32 = 131133;                                             // 131133 0x0002003D 131072 0x00020000
+const ERR_ACCOUNT_ID_INVALID_INSUFFICIENT_NUMBER_OF_ONES: u32 = 131132;                             // 131132 0x0002003C 
+const ERR_ACCOUNT_MUST_BE_A_FAUCET_TO_CALL_PROCEDURE: u32 = 131073;                                 // 131073 0x00020001
+const ERR_ACCOUNT_NONCE_INCREASE_MUST_BE_U32_VALUE: u32 = 131131;                                   // 131131 0x0002003B
+const ERR_ACCOUNT_POW_IS_INSUFFICIENT: u32 = 131135;                                                // 131135 0x0002003F
+const ERR_ACCOUNT_SEED_DIGEST_MISMATCH: u32 = 131134;                                               // 131134 0x0002003E
+
+// --- NONCE ERROR --------------------------------------------------------------------------------
+const ERR_NONCE_DID_NOT_INCREASE_AFTER_STATE_CHANGED: u32 = 131081;                                 // 131081 0x00020009
```

Comment:
This is probably should go into the "account errors" category as well.

Also, do we have the opposite error: "account nonce increased without state change"?

Reply:
>Also, do we have the opposite error: "account nonce increased without state change"?

For now we don't have this error and as far as I can see we don't check for this anywhere. Should I create a constant for this error?